frontend/src/types.ts

```diff
@@ -1516,6 +1516,8 @@ export interface InsightModel extends Cacheable {
     disable_baseline?: boolean
     filters: Partial<FilterType>
     query?: Node | null
+    /** Indicates that the results are tailored towards this dashboard and the insight is not relevant outside of it */
+    dashboardKey?: number
```

Comment:
This seems pretty dirty, because it's another "Only used in the frontend" property. Don't we have this info in logic props?

Reply:
Yes it's dirty. But the whole situation seems pretty chaotic.

On an insight itself, we have at most an array of dashboards where the insight appears. Which gets messy when the insight has results valid for one specific dashboard only. There is no other way, at least that I found, that tells us what dashboard the data is for.

For example: We seem to copy insights around pretty happily, not caring that the results are useless. When adding an insight to a dashboard from the detail view, we just plop it on the dashboard. It might be refreshed at that point just to be safe, but that just means that the single point of data will get overwritten and you end up with confusing results when you view it in another context.